src/Client/StorePaymentMethodLightningNetwork.php

```diff
@@ -0,0 +1,91 @@
+ */
+class StorePaymentMethodLightningNetwork extends AbstractStorePaymentMethodClient
+{
+    public function getPaymentMethods(string $storeId): array
+    {
+        $url = $this->getApiUrl() . 'stores/' . urlencode($storeId) . '/payment-methods/LightningNetwork';
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+        $response = CurlClient::request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            $r = [];
+            $data = json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+            foreach ($data as $item) {
+                $r[] = new \BTCPayServer\Result\StorePaymentMethodLightningNetwork($item, $item['cryptoCode'] . '-LightningNetwork');
+            }
+            return $r;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getPaymentMethod(string $storeId, string $cryptoCode): \BTCPayServer\Result\StorePaymentMethodLightningNetwork
+    {
+        $url = $this->getApiUrl() . 'stores/' . urlencode($storeId) . '/payment-methods/LightningNetwork/' . $cryptoCode;
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+        $response = CurlClient::request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            $data = json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+            return new \BTCPayServer\Result\StorePaymentMethodLightningNetwork($data, $data['cryptoCode'] . '-LightningNetwork');
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /**
+     * @param string $storeId
+     * @param string $cryptoCode
+     * @param array  $settings
+     *      Array of data to update. e.g.
+     *      [
+     *      'enabled' => true,
+     *      'connectionString' => 'Internal Node'
+     *      ]
+     *
+     * @return \BTCPayServer\Result\StorePaymentMethodLightningNetwork
+     * @throws \JsonException
+     */
+    public function updatePaymentMethod(string $storeId, string $cryptoCode, array $settings): \BTCPayServer\Result\StorePaymentMethodLightningNetwork
+    {
+        $url = $this->getApiUrl() . 'stores/' . urlencode($storeId) . '/payment-methods/LightningNetwork/' . $cryptoCode;
+        $headers = $this->getRequestHeaders();
+        $method = 'PUT';
+        $response = CurlClient::request($method, $url, $headers, json_encode($settings));
+
+        if ($response->getStatus() === 200) {
+            $data = json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+            return new \BTCPayServer\Result\StorePaymentMethodLightningNetwork($data, $data['cryptoCode'] . '-LightningNetwork');
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function removePaymentMethod(string $storeId, string $cryptoCode): bool
```

Comment:
Again I feel we need a comment. Deleting a payment method sounds very strange.

Reply:
That's why I named it remove like it does in the endpoint but as discussed on Mattermost it does not really delete/remove the payment method but at most the configured wallet/xpub. Will add something.

Reply:
Yeah, it's because of this discussion I remembered this. I did forget what the actual point was, so a comment here would help anyone who was not on Mattermost :)

Reply:
done